apps/authentication-service/src/resources/clients/clientsService.test.ts

```diff
@@ -0,0 +1,79 @@
+import { getUserClientId } from "@/shared/users/usersService";
+import { Client } from "auth0";
+import { http, HttpResponse } from "msw";
+
+import { AUTH0_USER_BY_ID } from "@/test/handlers";
+import {
+  exampleUser,
+  exampleToken,
+  exampleUserAlreadyHasAClientResponseError,
+  exampleUserInfoResponse,
+  exampleUserID,
+} from "@/test/testData/users";
+import { server } from "@/test/testServer";
+import {
+  exampleClient,
+  exampleClientDesc,
+  exampleClientName,
+} from "@/test/testData/clients";
+import { getClient, createClient, deleteClient } from "./clientsService";
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+jest.mock("auth0", () => ({
+  ...jest.requireActual("auth0"),
+  UserInfoClient: class {
+    getUserInfo = (_token: string) => ({ data: exampleUserInfoResponse });
+  },
+}));
+
+describe("Clients test", () => {
+  it("creates a new client", async () => {
+    server.use(
+      http.get(AUTH0_USER_BY_ID, () => HttpResponse.json(exampleUser)),
+    );
+    const client: Client = await createClient(exampleToken, {
+      name: exampleClientName,
+      description: exampleClientDesc,
+    });
+
+    expect(client).not.toBe(undefined);
+    expect(client).toEqual(exampleClient);
+  });
+
+  it("tries to create a new client when user already has one", async () => {
+    await expect(
+      createClient(exampleToken, {
+        name: exampleClientName,
+        description: exampleClientDesc,
+      }),
+    ).rejects.toEqual(exampleUserAlreadyHasAClientResponseError);
+  });
+
+  it("gets info for an existing client", async () => {
+    const client: Client = await getClient(exampleToken);
+
+    expect(client).not.toBe(undefined);
+    expect(client).toEqual(exampleClient);
+  });
+
+  it("deletes a client", async () => {
+    await deleteClient(exampleToken);
+
+    const response = await deleteClient(exampleToken);
+    expect(response).toBe(null);
+  });
+
+  it("deletes a client and check that the user_metadata has been cleared", async () => {
```

Comment:
We can't check that it modified the user's clientId in an integration test. Lines 72 and onward are faking all of that. You could and should test something like this in an e2e test(you might have already).

The best you could do in an integration test would be to check that the MSW handler for updating the client id was called with the correct payload or parameters